lib/common/formats.js

```diff
@@ -92,6 +92,47 @@ module.exports = {
       '\n}\n';
   },
 
+  /**
+   * Creates a Sass file with a flat map based on the style dictionary
+   *
+   * @memberof Formats
+   * @kind member
+   * @example
+   * ```scss
+   * $tokens: (
+   *   $color-background-base: #f0f0f0;
+   *   $color-background-alt: #eeeeee;
+   * )
+   * ```
+   */
+  'sass/map-flat': _.template(
+    fs.readFileSync(__dirname + '/templates/sass/map-flat.template')
+  ),
+
+  /**
+   * Creates a Sass file with a deep map based on the style dictionary
+   *
+   * @memberof Formats
+   * @kind member
+   * @example
+   * ```scss
+   * $color-background-base: #f0f0f0 !default;
+   * $color-background-alt: #eeeeee !default;
+   *
+   * $tokens: {
+   *   'color': (
+   *     'background': (
+   *       'base': $color-background-base,
+   *       'alt': $color-background-alt
+   *     )
+   *   )
+   * )
+   * ```
+   */
+  'sass/map-deep': _.template(
+    fs.readFileSync(__dirname + '/templates/sass/map-deep.template')
```

Comment:
I don't see this patter in any place or any file (or am I missing something?)

Reply:
Agreed @didoo. @dbanksdesign maybe we should create a new issue for this and do it across all templated formats?

Reply:
I actually don't think we should do it for the templated formats, rather for the normal functional formats that are currently written in formats.js. As we add more formats, that file becomes really long and it might make it more readable to make each format either it's own node module, or lodash template. I was also thinking if the formats in this PR might be more readable as a JS function rather than a lodash template because there is not a lot of template code, it is mostly functions and loops, but that is more personal preference.

Reply:
I agree with all the points you raised, @dbanksdesign. That said, I would postpone all these changes to later, with a proper PR to update the whole formats.js file as ou suggest.